compiler/rustc_ty_utils/src/abi.rs

```diff
@@ -436,10 +436,7 @@ fn fn_abi_sanity_check<'tcx>(
     ) {
         let tcx = cx.tcx();
 
-        if spec_abi == ExternAbi::Rust
-            || spec_abi == ExternAbi::RustCall
-            || spec_abi == ExternAbi::RustCold
-        {
+        if spec_abi.is_rustic_abi() {
```

Comment:
This is missing `RustIntrinsic`, but I think that's ok?

Reply:
yeah it was just checking that we don't put "weird" things into functions with these ABIs, so this just makes the check slightly stricter in a slightly irrelevant way because https://github.com/rust-lang/rust/issues/132735 is the death sentence of the weird "rust-intrinsic" pseudo-ABI anyways.